test_coo_numba::test_repack_attrs fails on 32bit due to type mismatch

**Describe the bug**
I am currently trying to fix and update the openSUSE package of python-sparse. There is one test failing on 32bit systems:

```
[  208s] E       numba.core.errors.LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
[  208s] E       Invalid store of [2 x i32] to [2 x i64] in <sparse._coo.numba_extension.COOModel object at 0xe76d4868> (trying to write member #2)
[  208s] E       
[  208s] E       File "sparse/tests/test_coo_numba.py", line 64:
[  208s] E               def pack(coords, data, shape):
[  208s] E                   return sparse.COO(coords, data, shape)
[  208s] E                   ^
[  208s] E       
[  208s] E       During: lowering "$12call_method.5 = call $4load_method.1(coords, data, shape, func=$4load_method.1, args=[Var(coords, test_coo_numba.py:64), Var(data, test_coo_numba.py:64), Var(shape, test_coo_numba.py:64)], kws=(), vararg=None)" at /home/abuild/rpmbuild/BUILD/sparse-0.11.2/sparse/tests/test_coo_numba.py (64)
[  208s] 
[  208s] /usr/lib/python3.8/site-packages/numba/core/utils.py:81: LoweringError
[  208s] =========================== short test summary info ============================
[  208s] FAILED sparse/tests/test_coo_numba.py::TestBasic::test_repack_attrs - numba.c...
[  208s] ====== 1 failed, 4310 passed, 29 xfailed, 36 xpassed in 134.16s (0:02:14) ======
```

**To Reproduce**
Build sparse on a system with 32bit libraries and run the unit test

**Expected behavior**
Unit tests pass / repacking 32bit type objects works

**System**
 - OS and version: Linux 32bit
 - `sparse` version 0.11.2
 - NumPy version 1.19.2
 - Numba version 0.51.2
- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1

**Additional context**
Full build log: 
[sparse-i586_log.txt](https://github.com/pydata/sparse/files/5741997/sparse-i586_log.txt)

--- a/sparse/_coo/numba_extension.py
+++ b/sparse/_coo/numba_extension.py
@@ -52,7 +52,7 @@ class COOType(types.Type):
 
     @property
     def shape_type(self):
-        return types.UniTuple(types.int64, self.ndim)
+        return types.UniTuple(types.intp, self.ndim)
 
     @property
     def fill_value_type(self):

--- a/sparse/_coo/numba_extension.py
+++ b/sparse/_coo/numba_extension.py
@@ -52,7 +52,8 @@ class COOType(types.Type):
 
     @property
     def shape_type(self):
-        return types.UniTuple(types.intp, self.ndim)
+        dt = numba.np.numpy_support.from_dtype(self.coords_dtype)
+        return types.UniTuple(at, self.ndim)
 
     @property
     def fill_value_type(self):

--- a/sparse/_coo/numba_extension.py
+++ b/sparse/_coo/numba_extension.py
@@ -53,7 +53,7 @@ class COOType(types.Type):
     @property
     def shape_type(self):
         dt = numba.np.numpy_support.from_dtype(self.coords_dtype)
-        return types.UniTuple(at, self.ndim)
+        return types.UniTuple(dt, self.ndim)
 
     @property
     def fill_value_type(self):

